src/proxy/topic/TopicProxy.ts

```diff
@@ -0,0 +1,123 @@
+import { ListenerMessageCodec} from '../../listener/ListenerMessageCodec';
+import {ClientMessage} from '../../protocol/ClientMessage';
+import {assertNotNull} from '../../util/Util';
+import {PartitionSpecificProxy} from '../PartitionSpecificProxy';
+import {TopicPublishCodec} from '../../codec/TopicPublishCodec';
+import Long = require('long');
+import {Data} from '../../serialization';
+import {ClientConfig, ClientConfigImpl} from '../../config';
+import {ProxyManager} from '../ProxyManager';
+import {PartitionService} from '../../PartitionService';
+import {InvocationService} from '../../invocation/InvocationService';
+import {SerializationService} from '../../serialization/SerializationService';
+import {ListenerService} from '../../listener/ListenerService';
+import {ClusterService} from '../../invocation/ClusterService';
+import {ConnectionRegistry} from '../../network/ConnectionRegistry';
+import {SchemaService} from '../../serialization/compact/SchemaService';
+import {Connection} from '../../network/Connection';
+import {TopicOverloadPolicy} from '../TopicOverloadPolicy';
+
+export class TopicProxy<E> extends PartitionSpecificProxy implements ITopic<E> {
+
+
+    private readonly overloadPolicy: TopicOverloadPolicy;
+    private readonly localAddress: AddressImpl;
+
+
+    constructor(
+        serviceName: string,
+        name: string,
+        clientConfig: ClientConfig,
+        proxyManager: ProxyManager,
+        partitionService: PartitionService,
+        invocationService: InvocationService,
+        serializationService: SerializationService,
+        listenerService: ListenerService,
+        clusterService: ClusterService,
+        connectionRegistry: ConnectionRegistry,
+        schemaService: SchemaService
+    ) {
+        super(
+            serviceName,
+            name,
+            proxyManager,
+            partitionService,
+            invocationService,
+            serializationService,
+            listenerService,
+            clusterService,
+            connectionRegistry,
+            schemaService
+        );
+        const connection: Connection = this.connectionRegistry.getRandomConnection();
+        this.localAddress = connection != null ? connection.getLocalAddress() : null;
+        const config = (clientConfig as ClientConfigImpl).getReliableTopicConfig(name);
+        this.overloadPolicy = config.overloadPolicy;
+    }
+
+    publish(message: E): Promise<void> {
+        assertNotNull(message);
+
+        const messageData = this.toData(message);
+        const partitionId = this.partitionService.getPartitionId(messageData);return this.encodeInvokeOnPartition(TopicPublishCodec, partitionId, () => {}, messageData);
+    }
+
+    publishAll(messages: any[]): Promise<void> {
+        assertNotNull(messages);
+        const messageDataList = this.toData(messages);
+        const partitionId = this.partitionService.getPartitionId(messageDataList);
```

Comment:
similar comment for this one about partition id.

Reply:
Done